src/adc.rs

```diff
@@ -365,6 +386,13 @@ macro_rules! adc_hal {
                     let res = self.rb.dr.read().data().bits();
                     res
                 }
+
+                /// Powers down the ADC, disables the ADC clock and releases the ADC Peripheral
+                pub fn release(mut self, apb2: &mut APB2) -> $ADC {
+                    self.power_down();
+                    self.disable_clock(apb2);
+                    self.rb
+                }
```

Comment:
Is there any reason why you don't want to return  $ADC from this function ?

Reply:
Well, no, but it is returning ADC. isn't it ?

Reply:
Oh, typo. Sorry for confusion. The question is "why do you want to return ADC from this function". I am asking not because I have any objections, I just want to understand the use-case.

Reply:
Because it is a singleton, I am returning it so people can use it to start the adc again if they want to.